x-pack/plugins/alerting/server/saved_objects/index.ts

```diff
@@ -217,6 +217,11 @@ export function setupSavedObjects(
   // Encrypted attributes
   encryptedSavedObjects.registerType({
     type: RULE_SAVED_OBJECT_TYPE,
+    /**
+     * We disable enforcing random SO IDs for the rule SO
+     * to allow users creating rules with a predefined ID.
+     */
+    enforceRandomId: false,
```

Comment:
I don't find anyone else in the code using this attribute(nor its implementation) where does this come from?

Reply:
We are the first ones! The security team implemented it for us. Check this issue: https://github.com/elastic/kibana/issues/194692.

Reply:
Gotcha, i was looking in a local branch that did not have their pr yet.